Shared/View/CreateActivityView.swift

```diff
@@ -46,9 +47,18 @@ struct CreateActivityView: View {
                     .font(.system(.title2))
                 ForEach(friendList, id: \.self) { item in
                     Text(item)
+                    
                 }
                 TextField("Name", text: $newFriend) {
                     self.friendList.append(self.newFriend)
+                    Task {
+                        do {
+                            // ERROR: a hard coded "username" works but when a user inputs an username saved through the self.newFriend variable, no id shows up
+                            try await self.friendID.append(UserViewModel.user(named: self.newFriend)?.id ?? "no id")
+                        } catch {
+                            print(error)
+                        }
+                    }
                     self.newFriend = ""
```

Comment:
Note that Task is executed later asynchronously, so by then newFriend is already empty string. You'll want to cache the newFriend to a local variable for task to use.